src/Agent.PluginCore/CommandPlugin.cs

```diff
@@ -0,0 +1,317 @@
+                }
+                else
+                {
+                    Credentials = new NetworkCredential(proxyUsername, proxyPassword);
+                }
+
+                if (proxyBypassList != null)
+                {
+                    foreach (string bypass in proxyBypassList)
+                    {
+                        if (string.IsNullOrWhiteSpace(bypass))
+                        {
+                            continue;
+                        }
+                        else
+                        {
+                            try
+                            {
+                                Regex bypassRegex = new Regex(bypass.Trim(), RegexOptions.IgnoreCase | RegexOptions.CultureInvariant | RegexOptions.ECMAScript);
+                                _regExBypassList.Add(bypassRegex);
+                            }
+                            catch (Exception)
+                            {
+                                // eat all exceptions
+                            }
+                        }
+                    }
+                }
+            }
+
+            public Uri GetProxy(Uri destination)
+            {
+                if (IsBypassed(destination))
+                {
+                    return destination;
+                }
+                else
+                {
+                    return new Uri(_proxyAddress);
+                }
+            }
+
+            public bool IsBypassed(Uri uri)
+            {
+                return string.IsNullOrEmpty(_proxyAddress) || uri.IsLoopback || IsMatchInBypassList(uri);
+            }
+
+            private bool IsMatchInBypassList(Uri input)
+            {
+                string matchUriString = input.IsDefaultPort ?
+                    input.Scheme + "://" + input.Host :
+                    input.Scheme + "://" + input.Host + ":" + input.Port.ToString();
+
+                foreach (Regex r in _regExBypassList)
+                {
+                    if (r.IsMatch(matchUriString))
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        private class CommandPluginClientCertificateManager : IVssClientCertificateManager
+        {
+            private readonly X509Certificate2Collection _clientCertificates = new X509Certificate2Collection();
+            public X509Certificate2Collection ClientCertificates => _clientCertificates;
+            public CommandPluginClientCertificateManager(string clientCertificateArchiveFile, string clientCertificatePassword)
+            {
+                if (!string.IsNullOrEmpty(clientCertificateArchiveFile))
+                {
+                    _clientCertificates.Add(new X509Certificate2(clientCertificateArchiveFile, clientCertificatePassword));
+                }
+            }
+        }
+
+        private AgentCertificateSettings GetCertConfiguration()
```

Comment:
is this consistent with how the libs read the settings? or how the agent reads the settings?